### PR TITLE
Restructure streaming tests to improve reliability

### PR DESCRIPTION
**Pull request by aron-cf**
We now create fewer sandboxes and only perform new sandbox creation at
the end of the test file. This will hopefully reduce the number of timeouts.

### CODE REVIEW

**Review by scuffi — APPROVED** [review]

File: tests/e2e/streaming-operations-workflow.test.ts [hunk 1]
```diff
@@ -281,7 +207,75 @@ describe('Streaming Operations Edge Cases', () => {
...
+        signal: AbortSignal.timeout(5000)
+      });
+      expect(stateResponse.status).toBe(200);
+      const state = (await stateResponse.json()) as { status: string };
+      status = state.status;
+      if (status === 'stopped' || status === 'stopped_with_code') {
+        break;
+      }
+      await new Promise((resolve) => setTimeout(resolve, 1000));
+    }
```
[the remark this answers is not in the document]
Comment (aron-cf) [reply tests/e2e/streaming-operations-workflow.test.ts:268]: I don't think this is true. The state api doesn't increment the timer. The tests pass because of this.
